operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -54,6 +62,13 @@ public void configureWith(KubernetesDependentResourceConfig<R> config) {
     this.kubernetesDependentResourceConfig = config;
   }
 
+  public void setMatcher(SSABasedGenericKubernetesResourceMatcher<R> matcher) {
+    if (this.matcher != defaultMatcher) {
```

Comment:
I'm not sure how big the matcher object can be, but this could be a boolean.

Reply:
Not sure I'm following

Reply:
I mean you don't need to keep the `defaultMatcher`in the field if it's only used to verify that the `matcher` is set only once. It can be a boolean that is flipped when first set so it's only taking a byte (usually) instead of the size of the `defaultMatcher` object.